supermarq-benchmarks/supermarq/benchmarks/phase_code.py

```diff
@@ -21,7 +21,12 @@ class PhaseCode(Benchmark):
 
     def __init__(self, num_data_qubits: int, num_rounds: int, phase_state: List[int]) -> None:
```

Comment:
The type check is failing because `mypy` is catching the type error before the specific `ValueError`'s are raised due to the type hint `List[int]`. Doing something like:  
```suggestion
    def __init__(self, num_data_qubits: int, num_rounds: int, phase_state: typing.Sized) -> None:
```
will make the type check pass by allowing `phase_state` to accept a broader range of input types (mainly objects that support `len()`) and then raising `ValueError`'s

Reply:
That makes sense, but I think it would be preferable if there was a way to keep the type in the function definition as `List[int]` -- this is clearest way to tell a user what the type should be, versus using a broader type requirement and then checking for what we want in the `__init__`.

I'm not sure if that is possible though

Reply:
Yeah, I thought that having `List[int]` would be more appropriate for the `__init__` and `typing.Sized` would be more appropriate for a general `validate_list_input()` function.

Looking around, it seems for our purposes, using something like `typing.cast` might get the job done for the tests while still keeping `List[int]`